libs/core/thread_support/include/hpx/thread_support/spinlock.hpp

```diff
@@ -22,7 +23,7 @@ namespace hpx { namespace util { namespace detail {
         HPX_NON_COPYABLE(spinlock);
 
     private:
-        std::atomic_flag m = ATOMIC_FLAG_INIT;
+        std::atomic<bool> m = false;
 
         HPX_CORE_EXPORT void yield_k(unsigned) noexcept;
```

Comment:
This is no longer needed.

Reply:
We might still need it after all as my commit created a circular dependency.  It might not be worth applying the optimization, after all...

Reply:
I peeked at that `yield_while` implementation, and it seems to be based in an older incarnation of `boost::smart_ptr::yield_k`. The one in this PR was based on the current `boost::smart_ptr::yield_k` implementation.

Reply:
Sure. The point of my commit was not replacing the `yield_k` you added (I'd be fine with changing our existing implementation to match what you proposed). The change I would like to see is to reduce the pressure on the atomic itself by not constantly trying to write to the atomic flag, but instead have a separate loop that just reads from it before trying to write again.